aws/table_aws_ecr_image.go

```diff
@@ -157,3 +165,25 @@ func listAwsEcrImages(ctx context.Context, d *plugin.QueryData, h *plugin.Hydrat
 
 	return nil, nil
 }
+
+func getImageURI(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	image := h.Item.(types.ImageDetail)
+	commonData, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	commonColumnData := commonData.(*awsCommonColumnData)
+
+	// AWS follows the below image URI format -
+	// with tag - {aws_account_id}.dkr.ecr.{region}.amazonaws.com/{repository name}:{first tag}
+	// without tag - {aws_account_id}.dkr.ecr.{region}.amazonaws.com/{repository name}@image_digest
+	if len(image.ImageTags) == 0 {
+		uri := commonColumnData.AccountId + ".dkr.ecr." + commonColumnData.Region + ".amazonaws.com/" + *image.RepositoryName + "@" + *image.ImageDigest
+
+		return uri, nil
+	}
+
+	uri := commonColumnData.AccountId + ".dkr.ecr." + commonColumnData.Region + ".amazonaws.com/" + *image.RepositoryName + ":" + image.ImageTags[0]
```

Comment:
@bigdatasourav Does AWS always take the first tag to form the URI? What happens if you use a different tag when building the URI and then pass that URI to other tools like Trivy?

Reply:
Yes, AWS always take the first tag to form the URI. Trivy can scan image URIs with other valid tags.